File: ruby/difference-of-squares/difference_of_squares.rb
class Squares

  attr_reader :number

  def initialize(number)
    @number = number
  end

  def square_of_sum
    (1..number).reduce(0, :+) ** 2
  end

  def sum_of_squares
    (1..number).reduce(0) do |sum, x|
      sum + (x ** 2)
    end
  end

  def difference
    square_of_sum - sum_of_squares
  end

end

module BookKeeping
  VERSION = 3
end

File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(x):
  return sum(range(1, x + 1)) ** 2

def sum_of_squares(x):
  return sum(y ** 2 for y in range(1, x + 1))

def difference(x):
  return square_of_sum(x) - sum_of_squares(x)
